lib/mappers/mongo.js

```diff
@@ -95,7 +95,13 @@ function MongoMapper () {
 
 				query[index] = instance[index];
 
-				return collection.updateAsync(query, instance);
+				return collection.findOneAsync(query)
```

Comment:
Is this really necessary? I would expect MongoDB to somehowe notify us that no documents were modified (however I recall we have had some issues with this). Maybe this could help: http://docs.mongodb.org/manual/reference/method/db.collection.update/#writeresult